trade-finance-manager-api/src/v1/controllers/auth/auth-provider.js

```diff
@@ -0,0 +1,183 @@
+const msal = require('@azure/msal-node');
+const axios = require('axios');
+const { msalConfig, TENANT_SUBDOMAIN, REDIRECT_URI, POST_LOGOUT_REDIRECT_URI } = require('./auth-provider-config');
+
+class AuthProvider {
+
+  constructor(config) {
+    this.config = config;
+    this.cryptoProvider = new msal.CryptoProvider();
+    this.msalConfig = msalConfig;
+  }
+
+  getMsalInstance() {
+    return new msal.ConfidentialClientApplication(this.msalConfig);
+  }
+
+  async getLoginUrl() {
+    // create a GUID for csrf
+    const csrfToken = this.cryptoProvider.createNewGuid();
+
+    /**
+     * The MSAL Node library allows you to pass your custom state as state parameter in the Request object.
+     * The state parameter can also be used to encode information of the app's state before redirect.
+     * You can pass the user's state in the app, such as the page or view they were on, as input to this parameter.
+     */
+
+    const state = this.cryptoProvider.base64Encode(
+      JSON.stringify({
+        csrfToken,
+        redirectTo: '/',
+      }),
+    );
+
+    const authCodeUrlRequestParams = {
+      state,
+
+      /**
+       * By default, MSAL Node will add OIDC scopes to the auth code url request. For more information, visit:
+       * https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
+       */
+      scopes: [],
```

Comment:
Should we add in the documentation, why we want `scopes` to be empty? I.e what happens if this is _not_ empty

Reply:
We have some scopes in Active Directory app setup, and it is working fine. Not sure if we need to add them here.

FYI this file is taken from Microsoft MSAL SSO example. MIT license.

Reply:
I've updated the documentation for this